cardinal/router/router.go

```diff
@@ -146,3 +152,14 @@ func (r *router) Start() error {
 	r.serverAddr = listener.Addr().String()
 	return nil
 }
+
+// intercepts grpc invocations and injects the router-key.
+func (r *router) clientCallInterceptor(
```

Comment:
This can be completely removed if you use the grpc.WithPerRPCCredentials option mentioned above.